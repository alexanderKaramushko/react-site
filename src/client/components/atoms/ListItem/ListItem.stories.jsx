import React from 'react';
import { storiesOf } from '@storybook/react';
import LabelItem, { ListItemSize } from './ListItem.component';
import { Themes } from '../../../common/settings';

storiesOf('LabelItem', module)
    .add('default', () => (
        <LabelItem>Test ListItem</LabelItem>
    ))
    .add('default - small', () => (
        <LabelItem size={ListItemSize.SMALL}>
            Test ListItem
        </LabelItem>
    ))
    .add('default - medium', () => (
        <LabelItem size={ListItemSize.MEDIUM}>
            Test ListItem
        </LabelItem>
    ))
    .add('default - large', () => (
        <LabelItem size={ListItemSize.LARGE}>
            Test ListItem
        </LabelItem>
    ))
    .add('theme - light', () => (
        <LabelItem theme={Themes.LIGHT}>
            Test ListItem
        </LabelItem>
    ))
    .add('theme - dark', () => (
        <LabelItem theme={Themes.DARK}>
            Test ListItem
        </LabelItem>
    ));
