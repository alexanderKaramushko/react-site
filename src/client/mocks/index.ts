import configureStore, { MockStoreEnhanced } from 'redux-mock-store';

export function mockStore<T>(initialState: T): MockStoreEnhanced {
  const configureMockStore = configureStore();
  const store = configureMockStore(initialState);

  return store;
}
