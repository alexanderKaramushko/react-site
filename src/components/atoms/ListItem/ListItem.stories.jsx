import React from 'react';
import { storiesOf } from '@storybook/react';
import LabelItem, { ListItemTheme, ListItemSize } from './ListItem.component';

storiesOf('LabelItem', module)
    .add('default', () => (
        <LabelItem>Test ListItem</LabelItem>
    ))
    .add('default - small', () => (
        <LabelItem size={ListItemSize.SMALL}>
            Test ListItem
        </LabelItem>
    ))
    .add('default - medium', () => (
        <LabelItem size={ListItemSize.MEDIUM}>
            Test ListItem
        </LabelItem>
    ))
    .add('default - large', () => (
        <LabelItem size={ListItemSize.LARGE}>
            Test ListItem
        </LabelItem>
    ))
    .add('theme - light', () => (
        <LabelItem theme={ListItemTheme.LIGHT}>
            Test ListItem
        </LabelItem>
    ))
    .add('theme - dark', () => (
        <LabelItem theme={ListItemTheme.DARK}>
            Test ListItem
        </LabelItem>
    ));
