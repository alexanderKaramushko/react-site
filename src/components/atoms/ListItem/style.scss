@use '../../_settings/mixins.scss';
@use '../../_settings/functions.scss';
@import '../../_settings/_variables';

/* Common */
.listItem {
    &:first-letter {
        color: functions.nested-map-get($colors, primary, primary_light);
    }
}

/* Size */
.small {
    padding: 10px;
    font-size: 14px;
}

.medium {
    padding: 15px;
    font-size: 16px;
}

.large {
    padding: 20px;
    font-size: 18px;
}

/* Theme */
.light, .dark a {
    @include mixins.colors('primary_dark', ('color'));
}

.dark, .dark a {
    @include mixins.colors('primary_light', ('color'));
}

a:hover {
    @include mixins.colors('error_light', ('color'));
}
