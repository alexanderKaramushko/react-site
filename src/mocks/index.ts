import configureStore, { MockStoreEnhanced } from 'redux-mock-store';

export function mockStore(initialState: any): MockStoreEnhanced {
    const configureMockStore = configureStore();
    const store = configureMockStore(initialState);

    return store;
}
