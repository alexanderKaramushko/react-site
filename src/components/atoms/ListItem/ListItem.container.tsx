import { connect } from 'react-redux';
import { RootState } from '../../../store/reducers/rootReducer';
import { getActiveTheme } from '../../../store/reducers/settings/selectors';
import { StateProps, DispatchProps, OwnProps } from './ListItem.types';
import ListItem from './ListItem.component';

export default connect<StateProps, DispatchProps, OwnProps, RootState>(
    (state) => ({
        theme: getActiveTheme(state),
    }),
)(ListItem);
