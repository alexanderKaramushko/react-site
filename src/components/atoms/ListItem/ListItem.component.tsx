import React from 'react';
import classnames from 'classnames';
import * as styles from './style.scss';
import { Themes } from '../../../common/settings';
import { OwnProps } from './ListItem.types';

export const ListItemTheme = {
    DARK: Themes.DARK,
    LIGHT: Themes.LIGHT,
};

export const ListItemSize = {
    LARGE: 'large',
    MEDIUM: 'medium',
    SMALL: 'small',
};

const defaultProps = {
    children: '',
    size: 'medium',
    theme: Themes.LIGHT,
};

const ListItem: React.FunctionComponent<OwnProps> = (props) => {
    const { children, size, theme } = props;
    const classProps = classnames(styles.listItem, styles[theme], styles[size]);

    return (
        <li className={classProps}>
            {children}
        </li>
    );
};

ListItem.defaultProps = defaultProps;
ListItem.displayName = 'ListItem';

export default ListItem;
