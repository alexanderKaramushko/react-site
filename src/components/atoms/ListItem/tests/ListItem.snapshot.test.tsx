import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MockStoreEnhanced } from 'redux-mock-store';
import { mockStore } from '../../../../mocks';
import ListItem from '../ListItem.container';

const store: MockStoreEnhanced = mockStore({
    settingsReducer: {
        themeName: 'light',
    },
});

describe('>>>ListItem --- Snapshot', () => {
    it('+++capturing Snapshot of ListItem', () => {
        const renderedValue = renderer.create(
            <Provider store={store}>
                <ListItem>
                    <span>Test ListItem</span>
                </ListItem>
            </Provider>,
        ).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
});
