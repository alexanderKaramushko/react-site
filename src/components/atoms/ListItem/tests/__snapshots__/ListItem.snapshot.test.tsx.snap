// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>ListItem --- Snapshot +++capturing Snapshot of ListItem 1`] = `
<li
  className=""
>
  <span>
    Test ListItem
  </span>
</li>
`;
